### PR TITLE
Stop harcoding RK value sent to the authenticator

### PR DESCRIPTION
**Pull request by coroiu**
UP, UV and RK are currently hardcoded to true in the client, meaning that we will always create discoverable credentials, even if they aren’t actually requested.

----------------------------

Bitwarden internal tracking:
[PM-8578] based on [PM-7150]

### CODE REVIEW

File: passkey-client/src/lib.rs [hunk 1]
```diff
@@ -367,6 +364,47 @@ where
...
+                resident_key: Some(ResidentKeyRequirement::Preferred),
+                ..
+            // > Let requireResidentKey be true.
+            } => true,
+
+            // > is present and set to preferred
+            // >  And is not capable of client-side credential storage modality, or if the client cannot determine authenticator capability,
+            // >  Let requireResidentKey be false.
+            // `passkey-rs` requires the authenticator to be capable of client-side credential storage modality,
+            // so we do not need to check for this case.
```
Comment (Progdrasil) [inline passkey-client/src/lib.rs:390]: **Suggestion:** We can pass in the result of `authenticator.get_info` and look at `get_info::Response::options::rk` for this

Comment (coroiu) [reply passkey-client/src/lib.rs]: Something like this? :)

File: passkey-client/src/lib.rs [hunk 2]
```diff
@@ -367,6 +364,45 @@ where
             client_extension_results: AuthenticatorExtensionsClientOutputs::default(),
         })
     }
+
+    fn map_rk(&self, criteria: &Option<AuthenticatorSelectionCriteria>) -> bool {
+        let supports_rk = self.authenticator.get_info().options.is_some_and(|o| o.rk);
```
Comment (Progdrasil) [inline passkey-client/src/lib.rs:369]: **Suggestion:** This method doesn't need to call `get_info`. It's already done in the `register` method, the result should just be passed down to this method instead.

Comment (coroiu) [reply passkey-client/src/lib.rs]: Fixed!

**Review by Progdrasil — APPROVED** [review]

**Pull request merged**